File: app/src/main/java/tw/com/louis383/coffeefinder/list/ListPresenter.java
package tw.com.louis383.coffeefinder.list;

import java.util.List;

import tw.com.louis383.coffeefinder.BasePresenter;
import tw.com.louis383.coffeefinder.model.domain.CoffeeShop;

/**
 * Created by louis383 on 2017/2/21.
 */

public class ListPresenter extends BasePresenter<ListPresenter.ViewHandler> {

    public void prepareToShowCoffeeShops(List<CoffeeShop> coffeeShops) {
        if (!coffeeShops.isEmpty()) {
            view.setItems(coffeeShops);
            view.setRecyclerViewVisibility(true);
        } else {
            view.showNoCoffeeShopMessage();
            view.setNoCoffeeShopPictureVisibility(true);
        }
        view.setLoadingProgressBarVisibility(false);
    }

    public interface ViewHandler {
        void setLoadingProgressBarVisibility(boolean visible);
        void setRecyclerViewVisibility(boolean visible);
        void setNoCoffeeShopPictureVisibility(boolean visible);
        void showNoCoffeeShopMessage();
        void setItems(List<CoffeeShop> items);
    }
}

File: app/src/main/java/tw/com/louis383/coffeefinder/list/ListPresenter.kt
package tw.com.louis383.coffeefinder.list

import android.location.Location
import dagger.hilt.android.scopes.FragmentScoped
import tw.com.louis383.coffeefinder.BasePresenter
import tw.com.louis383.coffeefinder.core.CurrentLocationCarrier
import tw.com.louis383.coffeefinder.core.domain.model.CoffeeShop
import javax.inject.Inject

/**
 * Created by louis383 on 2017/2/21.
 */

@FragmentScoped
class ListPresenter @Inject constructor(private val currentLocationCarrier: CurrentLocationCarrier) : BasePresenter<CoffeeShopListView>() {

    fun prepareToShowCoffeeShops(coffeeShops: List<CoffeeShop>) {
        if (coffeeShops.isNotEmpty()) {
            view?.setItems(coffeeShops)
            view?.setRecyclerViewVisibility(true)
        } else {
            view?.showNoCoffeeShopMessage()
            view?.setNoCoffeeShopPictureVisibility(true)
        }
        view?.setLoadingProgressBarVisibility(false)
    }

    fun getCurrentLocation(): Location? {
        return currentLocationCarrier.currentLocation
    }
}
